service/controller/clusterapi/v17/resources/clusterstatus/create.go

```diff
@@ -2,8 +2,242 @@ package clusterstatus
 
 import (
 	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	"github.com/giantswarm/errors/tenant"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/tenantcluster"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/label"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/controllercontext"
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring cluster status is up to date")
+
+	currentStatus := r.commonClusterStatusAccessor.GetCommonClusterStatus(cr)
+
+	updatedStatus, err := r.ensureClusterHasID(ctx, cr, currentStatus)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Ensure that cluster has cluster ID label.
+	cr.Labels[label.Cluster] = updatedStatus.ID
+
+	var nodes []corev1.Node
+	var tenantAPIAvailable bool
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding nodes of tenant cluster")
+
+		l, err := cc.Client.TenantCluster.K8s.CoreV1().Nodes().List(metav1.ListOptions{})
+		if tenantcluster.IsTimeout(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "timeout fetching certificates")
```

Comment:
You should return the error after logging.

Reply:
We should not log and return errors. This is an anti pattern as the same information gets managed twice. When you operate a service this is pretty confusing IMO. The error here is expected and we do it like this in a lot of cases. This is fine.

Reply:
Yes, both of these two cases are such that overall system behaviour would be in dead lock if there was return here. When cluster is created, it's expected that there are no certs (because this resource gets executed before Cluster -> AWSClusterConfig mapping resource and certs are created from AWSClusterConfig) or that API is down (because there were no secrets, certs or even cluster ID in status so that aws-operator could create the cluster).

TL;DR; As Tim already commented above, this is fine. :slightly_smiling_face: